src/Core/Module/ModuleManager.php

```diff
@@ -0,0 +1,355 @@
+            throw new Exception($this->translator->trans(
+                'You are not allowed to enable the module %module% on mobile.',
+                ['%module%' => $name],
+                'Admin.Modules.Notification'
+            ));
+        }
+
+        $this->assertIsInstalled($name);
+
+        $this->hookManager->exec('actionBeforeEnableMobileModule', ['moduleName' => $name]);
+
+        $module = $this->moduleRepository->getModule($name);
+        $enabled = $module->onMobileEnable();
+        $this->dispatch(ModuleManagementEvent::ENABLE_MOBILE, $module);
+
+        return $enabled;
+    }
+
+    public function disableMobile(string $name): bool
+    {
+        if (!$this->adminModuleDataProvider->isAllowedAccess(__FUNCTION__, $name)) {
+            throw new Exception($this->translator->trans(
+                'You are not allowed to disable the module %module% on mobile.',
+                ['%module%' => $name],
+                'Admin.Modules.Notification'
+            ));
+        }
+
+        $this->assertIsInstalled($name);
+
+        $this->hookManager->exec('actionBeforeDisableMobileModule', ['moduleName' => $name]);
+
+        $module = $this->moduleRepository->getModule($name);
+        $disabled = $module->onMobileDisable();
+        $this->dispatch(ModuleManagementEvent::DISABLE_MOBILE, $module);
+
+        return $disabled;
+    }
+
+    public function reset(string $name, bool $keepData = false): bool
+    {
+        if (
+            !$this->adminModuleDataProvider->isAllowedAccess('install') ||
+            !$this->adminModuleDataProvider->isAllowedAccess('uninstall', $name)
+        ) {
+            throw new Exception($this->translator->trans(
+                'You are not allowed to reset the module %module%.',
+                ['%module%' => $name],
+                'Admin.Modules.Notification'
+            ));
+        }
+
+        $this->assertIsInstalled($name);
+
+        $this->hookManager->exec('actionBeforeResetModule', ['moduleName' => $name]);
+
+        $module = $this->moduleRepository->getModule($name);
+
+        if ($keepData && method_exists($module, 'reset')) {
+            $reset = $module->onReset();
+            $this->dispatch(ModuleManagementEvent::RESET, $module);
+        } else {
+            $reset = $this->uninstall($name) && $this->install($name);
+        }
+
+        return $reset;
+    }
+
+    public function isInstalled(string $name): bool
+    {
+        return $this->moduleDataProvider->isInstalled($name);
+    }
+
+    public function isEnabled(string $name): bool
+    {
+        return $this->moduleDataProvider->isEnabled($name);
+    }
+
+    public function getError(string $name): string
+    {
+        $module = $this->moduleRepository->getModule($name);
+        if ($module->hasValidInstance()) {
+            $errors = $module->getInstance()->getErrors();
+            $error = array_pop($errors);
+            if (empty($error)) {
```

Comment:
Or filter all empty errors to keep only interesting messages
```suggestion
            $errorMessages = array_filter($module->getInstance()->getErrors());

            if (empty($errorMessages)) {
```